cpp/include/cudf/detail/groupby/group_shift.hpp

```diff
@@ -0,0 +1,47 @@
+/*
+ * Copyright (c) 2021, NVIDIA CORPORATION.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include <cudf/groupby.hpp>
+#include <cudf/types.hpp>
+#include <cudf/utilities/span.hpp>
+
+#include <rmm/cuda_stream_view.hpp>
+
+#include <memory>
+#include <utility>
+
+namespace cudf {
+namespace groupby {
+namespace detail {
+
+/**
+ * @copydoc cudf::groupby::groupby::shift(column_view const&, size_type, scalar const&,
+ * rmm::mr::device_memory_resource)
+ *
+ * @param sorted_values values to be shifted, sorted by keys
+ * @param stream CUDA stream used for device memory operations and kernel launches.
+ */
+std::unique_ptr<column> group_shift(
```

Comment:
Instead of calling this `group_shift` I wonder if we should call this a `segmented_shift`, since we could presumably use this to shift values in a list column as well?

Reply:
`segmented_shift` is more generic name for underlying operation of this function.